cloudtrust-common/src/main/java/io/cloudtrust/crypto/CryptoUtil.java

```diff
@@ -125,10 +133,15 @@ public static String gcmDecrypt(SecretKey aesKey, String encryptedText) throws B
      */
     public static String encryptForDatabaseStorage(byte[] data) {
         try {
-            byte[] dbEncryptionKey = getCtDatabaseSecurityKey();
-            SecretKey originalKey = new SecretKeySpec(dbEncryptionKey, 0, dbEncryptionKey.length, "AES");
-            return gcmEncryptData(originalKey, data);
-        } catch (BadPaddingException | IllegalBlockSizeException ex) {
+            KeyEntry dbEncryptionKey = getCtDatabaseSecurityKey();
+            String encData = Base64.getEncoder().encodeToString(data);
+            if (dbEncryptionKey.value.length > 0) {
+                SecretKey originalKey = new SecretKeySpec(dbEncryptionKey.value, 0, dbEncryptionKey.value.length, "AES");
```

Comment:
Would be good to directly return this SecretKey object to avoid to perform this operation for each encryption

Reply:
fixed